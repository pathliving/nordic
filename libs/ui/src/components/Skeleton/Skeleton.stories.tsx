import type { Meta, StoryObj } from "@storybook/react";
import Skeleton from "./Skeleton";

const meta: Meta<typeof Skeleton> = {
  title: "Components/Skeleton",
  component: Skeleton,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
};
export default meta;

type Story = StoryObj<typeof Skeleton>;

const ArticleTemplate = () => {
  return (
    <div style={{ display: 'flex', flexDirection: 'column', gap: 'var(--s16)', borderColor: 'var(--secondary)', height: '140px', width: '400px' }}>
      <div style={{ display: 'flex', gap: 'var(--s16)' }}>
        <div style={{ flex: 'none' }}>
          <Skeleton appearance="circle" />
        </div>
        <div style={{ display: 'flex', flexDirection: 'column', gap: 'var(--s16)', width: '100%' }}>
          <Skeleton appearance="text" />
          <Skeleton appearance="text" />
        </div>
      </div>
      <Skeleton appearance="text" />
      <Skeleton appearance="text" />
    </div>
  )
}

export const Container: Story = (args: Story["args"]) => {
  return (
    <figure style={{ width: '450px', height: '200px' }}>
      <Skeleton appearance={args?.appearance ?? "text"} {...args} />
      <figcaption>assets/not-found-image.svg</figcaption>
    </figure>
  );
};
Container.args = {
  className: 'skeleton-container',
  appearance: "container",
}
Container.storyName = 'Image Skeleton';

export const ArticleSkeleton = () => {
  return (
    <ArticleTemplate />
  );
};
Container.storyName = 'Article Skeleton';
