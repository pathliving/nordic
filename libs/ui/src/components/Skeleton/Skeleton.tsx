import clsx from 'clsx';
import { DetailedHTMLProps, forwardRef, HTMLAttributes } from 'react';
import style from './skeleton.module.css';

import '@/styles/colors.css';
import '@/styles/spacing.css';
import '@/styles/transition.css';

export type Props = DetailedHTMLProps<
  HTMLAttributes<HTMLSpanElement>,
  HTMLSpanElement
> & {
  /**
   * What additional styles to add
   */
  className?: string;
  /**
   * What shape of skeleton to choose
   */
  appearance?: 'container' | 'circle' | 'text';
};

/**
 * Skeleton loader
 */
const Skeleton = forwardRef<HTMLSpanElement, Props>((props, ref) => {
  const { className, appearance = 'container', ...rest } = props;

  return (
    <span
      className={clsx(style.skeleton, style[appearance], className)}
      ref={ref}
      {...rest}
    />
  );
});

export default Skeleton;
