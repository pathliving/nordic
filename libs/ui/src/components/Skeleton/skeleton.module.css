.skeleton {
  background-color: var(--secondary);
  display: inline-flex;
  position: relative;
  pointer-events: none;
  user-select: none;
}

.skeleton::before {
  content: "";
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 0;
  height: 100%;
  box-shadow: 0 0 150px 50px var(--white);
  animation: wave 1s ease-in-out infinite;
}

.container {
  width: 100%;
  height: 100%;
  border-radius: var(--s8);
}

.circle {
  width: var(--s48);
  height: var(--s48);
  border-radius: 50%;
}

.text {
  width: 100%;
  height: var(--s20);
  border-radius: var(--s4);
}

@keyframes wave {
  from {
    left: -100px;
  }
  to {
    left: calc(100% + 100px);
  }
}