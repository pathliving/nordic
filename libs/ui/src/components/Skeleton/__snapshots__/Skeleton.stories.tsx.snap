// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Skeleton ArticleSkeleton smoke-test 1`] = `
<div style="display: flex; flex-direction: column; gap: var(--s16); border-color: var(--secondary); height: 140px; width: 400px;">
  <div style="display: flex; gap: var(--s16);">
    <div style="flex: 0 0 auto;">
      <span class="_skeleton_15e9e_1 _circle_15e9e_27">
      </span>
    </div>
    <div style="display: flex; flex-direction: column; gap: var(--s16); width: 100%;">
      <span class="_skeleton_15e9e_1 _text_15e9e_33">
      </span>
      <span class="_skeleton_15e9e_1 _text_15e9e_33">
      </span>
    </div>
  </div>
  <span class="_skeleton_15e9e_1 _text_15e9e_33">
  </span>
  <span class="_skeleton_15e9e_1 _text_15e9e_33">
  </span>
</div>
`;

exports[`Components/Skeleton Container smoke-test 1`] = `
<figure style="width: 450px; height: 200px;">
  <span class="_skeleton_15e9e_1 _container_15e9e_21 skeleton-container">
  </span>
  <figcaption>
    assets/not-found-image.svg
  </figcaption>
</figure>
`;
